### ISSUE TITLE
Chart test image missing.

### ISSUE THREAD

**Issue by trombka** [eshepelyuk/apicurio-registry-helm#4]
Hi,

The chart's test fails due to the lack of a Docker image configured in the `e2e.yaml` resource manifest.

I thought about using this chart as a dependency in two I maintain. But the failing test prevents me from doing so because there is no way in Helm to disable tests from a single dependency.

Could please you either provide the missing image or drop the test resource if it is just a junk?

**Comment by eshepelyuk** [comment]
Helm test is only for CI, don't run it in real cluster.

**Comment by trombka** [comment]
Hmm... No, it is not.

Helm test is to (according to the documentation) "validate that your chart works as expected when it is installed".

And the problem with Helm is that you cannot disable a test for a dependency. So if I put a dependency into my chart that has a broken test, I cannot run after-installation tests for my chart.

**Comment by trombka** [comment]
It is not for a chart's author, but a chart's user.

**Comment by eshepelyuk** [comment]
1. Helm test of this helm chart is not intented for any usage outside CI.
1. I will not expose the test image to the public, because of previous point.
1. Afaik, Helm tests are only executed when someone calls `helm test`, but I will double check this.
1. From the results of previous point I will either report results here or provide bugfixes.

wdyt ?

**Comment by trombka** [comment]
True for no.3, but not only when it is called on your chart, but on any chart depending on yours.
That's why you cannot make no.1 assumption.
 If you intend to follow no.2. you need to remove the test from the published chart. 

Please do the following:
```
$ helm create apicurio-registry-depender
$ cd apicurio-registry-depnder
$ helm install tmp .
$ helm test tmp
```
You should get a successful test result of the installation of the sample chart.
Now add a dependency to your chart by adding at the end of `Charts.yaml`:
```
dependencies:
  - name: apicurio-registry
    version: 1.3.0
    repository: "oci://ghcr.io/eshepelyuk/"
```
And do:
```
$ export HELM_EXPERIMENTAL_OCI=1
$ helm dependencies update
$ helm upgrade tmp .
$ helm test tmp
```
You will get failed test result and the last command will return ERR status (which stops further execution of a shell script).

In short: Adding your chart as a dependency breaks the testing of installation of any depending chart.

**Comment by eshepelyuk** [comment]
Hmmmmmm 
Are you directly calling `helm test` ??? After the installation of your umbrella chart ??

**Comment by trombka** [comment]
Yes. Exactly following the script above.
`helm test` always runs tests of dependencies as well and AFAIK there is no way to disable it.

**Comment by eshepelyuk** [comment]
I am not gonna address this, because `helm test` is never used during chart installation/upgrade. Instead chart installation/upgrade success is determined by liveness/readiness/startup probes.

Is you consider that chart is missing any of those probes - please open an issue or provide a PR.

To fix your own problem - just sont run`helm test`.

**Comment by trombka** [comment]
How did you come to a conclusion that `helm test` is intended for only your CI process and not any CI process of your chart consumers?

**Comment by eshepelyuk** [comment]
This dscussion is closed. No more replies gonna be provided. I've described possible options above. Please consider either of them. 
Thanks a lot.

**Comment by trombka** [comment]
It's sad. You gave no option to test a depending chart and it makes your valuable work here unusable. 
It seems a simple fix just to remove the failing test from the published artefact in Artifacthub. If you don't want it to be executed why do you publish it then?
